src/main/java/org/greenbuttonalliance/gbaresourceserver/usage/model/IntervalBlock.java

```diff
@@ -45,4 +47,8 @@ public class IntervalBlock extends IdentifiedObject {
 
 	@OneToMany(mappedBy = "block", cascade = CascadeType.ALL)
 	private Set<IntervalReading> intervalReadings = new HashSet<>();
+
+	@ManyToOne(optional = false, cascade = {CascadeType.PERSIST, CascadeType.MERGE})
```

Comment:
Just in case it's not clear why this isn't just `CascadeType.ALL`, unlike `@OneToMany`s, `@ManyToOne`s typically shouldn't cascade delete operations; for example, someone might want to delete a single `IntervalBlock` from a `MeterReading` without deleting the entire `MeterReading`. (By contrast, if someone deletes an entire `MeterReading`, all its `IntervalBlock`s should be dropped as well.) However, cascading creation/update operations can occasionally be useful (e.g., in our repository tests), so we want to cascade `PERSIST` and `MERGE` here without cascading `ALL`.

Reply:
Excellent solution!